questionnaires/questionnaires.rst

```diff
@@ -177,83 +345,262 @@ else).
+        unique characters.
+
+        aio
+
+  .. tab-content:: tab-code-freetext
+    :title: RST Code
+
+    .. code-block:: rst
+
+      .. questionnaire:: questionnaire_text_demo_2 10
+
+        .. freetext:: 5 string-ignorews-ignorequotes
+          :length: 10
+
+          Here the correct answer is ``anothertest``. Surrounding quotes are ignored in the
+          solution as well as whitespace everywhere. (modifiers ignorequotes and ignorews).
+
+          anothertest
+          !anothertest § Check the correct answer given in the description
+
+        .. freetext:: 5 unsortedchars-ignorews
+          :length: 7
+
+          An ``unsortedchars`` example. What are the unique vovels in the word
+          "cacophonic"? Correct answers are: aio, aoi, iao, ioa, oai, oia, and
+          also the versions with two o's, because *unsortedchars* always compares
+          unique characters.
+
+          aio
+
+Example: Regex questionnaire
+............................
+
+.. rst-tabs::
+
+  .. tab-content:: tab-html-render-regexp
+    :title: HTML visualisation
+
+    .. questionnaire:: questionnaire_regexp 20
+      :title: Questionnaire using regular expressions
+      :submissions: 5
+
+      Regular expressions are useful when there are multiple solutions, or when
+      one wants to have some tolerance in numeric questions, like accept real
+      numbers beginning with 0.014, 0.015, or 0.016.
+
+      .. freetext:: 10 regexp
+        :length: 7
+
+        Type either "cat" or "dog".
+
+        ^(cat|dog)$
+
+      .. freetext:: 10 regexp
+        :length: 7
+
+        What is the value of :math:`\pi` with four most significant digits?
+        This will accept ``3.141``, ``3.1415``, ``3.1416``, ``3.14159``, that is,
+        ``3.141`` and zero or more digits after that.
+
+        ^3\.141\d*$
+
+  .. tab-content:: tab-code-regexp
+    :title: RST Code
+
+    .. code-block:: rst
+
+      .. questionnaire:: questionnaire_regexp 20
+        :title: Questionnaire using regular expressions
+        :submissions: 5
+
+        Regular expressions are useful when there are multiple solutions, or when
+        one wants to have some tolerance in numeric questions, like accept real
+        numbers beginning with 0.014, 0.015, or 0.016.
+
+        .. freetext:: 10 regexp
+          :length: 7
+
+          Type either "cat" or "dog".
+
+          ^(cat|dog)$
+
+        .. freetext:: 10 regexp
+          :length: 7
+
+          What is the value of :math:`\pi` with four most significant digits?
+          This will accept ``3.141``, ``3.1415``, ``3.1416``, ``3.14159``, that is,
+          ``3.141`` and zero or more digits after that.
+
+          ^3\.141\d*$
+
+Testing the questionnaires
+--------------------------
+
+It is good practise to test your questionnaire, especially if there are
+several correct answers. Note that A+ will show the correct answers for the
+students who have submitted for the maximum number of times (but not for anyone
```

Comment:
The comment about the questionnaire showing the correct answers after reaching the submission limit could emphasize that it can be disabled and it is disabled by default. The questionnaire option `reveal-model-at-max-submissions` is listed at the start of this chapter.

Reply:
I would prefer to remove the whole Testing questionnaire section. It adds no value to this chapter. On the other hand, the `revel-model-at-max-submissions` option is explained above, so adding that information here is not relevant.